### PR TITLE
Add ENVHandler class to handle env and runtime injection

### PR DESCRIPTION
**Pull request by cosmosgenius**
- Adds a hard check on ENV access by irene.
- Adds a unified way to retrieve ENV from both `process.env` and `runtimeConfig`
- Adds a function to get plugin values from ENV with dependency on `Enterprise`

### CODE REVIEW

File: app/initializers/irene.js [hunk 1]
```diff
@@ -1,25 +1,91 @@
...
+  constructor(envHandlerConst) {
+    this.envHandlerConst = envHandlerConst;
+  }
 
-function isTrue(value) {
-  value = String(value).toLowerCase();
-  return value === 'true';
-}
+  isRuntimeAvailable() {
+    return !(typeof runtimeGlobalConfig == "undefined");
```
Comment (guganabu) [inline app/initializers/irene.js:9]: how does this variable becomes accessible?

File: config/environment.js [hunk 2]
```diff
@@ -1,71 +1,201 @@
...
+  ],
+  processENV: Object.keys(process.env).reduce((acc, key) => {
+    if (possibleENVS.indexOf(key) > -1) {
+      acc[key] = process.env[key];
+    }
+    return acc;
+  }, {}),
+};
+
+class ENVHandler {
```
Comment (guganabu) [inline config/environment.js:62]: both initializer and env has the same ENVHandler class implementation, is that required to be separated?

File: config/environment.js [hunk 3]
```diff
@@ -103,267 +233,429 @@ module.exports = function(environment) {
...
-      routeIfAlreadyAuthenticated: 'authenticated.index',
+    "ember-simple-auth": {
+      authenticationRoute: "/login",
+      loginEndPoint: "/login",
+      checkEndPoint: "/check",
+      logoutEndPoint: "/logout",
+      routeAfterAuthentication: "authenticated.index",
+      routeIfAlreadyAuthenticated: "authenticated.index",
     },
     endpoints: {
```
Comment (guganabu) [inline config/environment.js:309]: is this really active endpoints?

**Review by guganabu — COMMENTED** [review]
Please take a look into the comments which I have added

File: app/initializers/irene.js (hunk 1 above)
Comment (cosmosgenius) [reply app/initializers/irene.js:9]: runtimeGlobalConfig is injected via a script as global variable. See runtimeconfig.js

File: config/environment.js (hunk 2 above)
Comment (cosmosgenius) [reply config/environment.js:62]: nope, i have no idea how to unify them. I need ENVHandler in both environment.js and in initializer, but didn't find a way to share it from a single place.

**Review by guganabu — APPROVED** [review]

**Review by Praseetha-KR — APPROVED** [review]

**Pull request merged**